src/Neo/SmartContract/Native/NativeContract.cs

```diff
@@ -379,8 +383,8 @@ internal async void Invoke(ApplicationEngine engine, byte version)
                     throw new InvalidOperationException($"Cannot call this method before hardfork {method.ActiveIn}.");
                 if (method.DeprecatedIn is not null && engine.IsHardforkEnabled(method.DeprecatedIn.Value))
                     throw new InvalidOperationException($"Cannot call this method after hardfork {method.DeprecatedIn}.");
-                ExecutionContextState state = context.GetState<ExecutionContextState>();
-                if (!state.CallFlags.HasFlag(method.RequiredCallFlags))
+                var state = context.GetState<ExecutionContextState>();
+                if (!method.CheckCallFlags(engine, state.CallFlags))
```

Comment:
While this should work, I'm worried about the level of complexity and duplication we add here. Ideally I'd like to see proper per-hardfork manifests being used for invocations, this at least would eliminate the need for `RegisterAndUnregisterCallflags` (everything can be conveyed at `CreateMethodMetadataInternal` stage).

Reply:
Trying a new way